modules/stochastic_tools/include/utils/CovarianceFunction.h

```diff
@@ -0,0 +1,139 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "libmesh/utility.h"
+
+#include "DataIO.h"
+#include "MooseEnum.h"
+
+class GaussianProcessTrainer;
+
+namespace CovarianceFunction
+{
+class CovarianceKernel;
```

Comment:
"Kernel" should be changed to something else so this doesn't get confused with a MOOSE Kernel

Reply:
Done, calling these Covariance Function in general.